src/Functions/LeastGreatestGeneric.h

```diff
@@ -111,7 +111,7 @@ class LeastGreatestOverloadResolver : public IFunctionOverloadResolver
             argument_types.push_back(argument.type);
 
         /// More efficient specialization for two numeric arguments.
-        if (arguments.size() == 2 && isNumber(arguments[0].type) && isNumber(arguments[1].type))
+        if (arguments.size() == 2 && isNumber(removeNullable(arguments[0].type)) && isNumber(removeNullable(arguments[1].type)))
```

Comment:
There is another condition like this in `getReturnTypeImpl` which I guess needs to be adjusted to match

Reply:
done